rule/fetcher_default_test.go

```diff
@@ -252,11 +289,11 @@ access_rules:
 		{content: "[]"},
 		{content: `[{"id":"1"}]`, expectIDs: []string{"1"}},
 		{content: `[{"id":"1"},{"id":"2"}]`, expectIDs: []string{"1", "2"}},
-		{content: `[{"id":"2"},{"id":"3"}]`, expectIDs: []string{"2", "3"}},
+		{content: `[{"id":"2"},{"id":"3"},{"id":"4"}]`, expectIDs: []string{"2", "3", "4"}},
 	} {
 		t.Run(fmt.Sprintf("case=%d", k), func(t *testing.T) {
-			require.NoError(t, ioutil.WriteFile(repository, []byte(tc.content), 0777))
-			time.Sleep(time.Millisecond * 500)
+			repoFile.Truncate(0)
+			repoFile.WriteAt([]byte(tc.content), 0)
```

Comment:
Does this flush automatically? If not we could force a flush which might increase test speed

Reply:
I added syncs.